### PR TITLE
Fixes #14398

### LINKED ISSUE #14398: Add support of the UV_THREADPOOL_SIZE
### What is the problem this feature would solve?

Add support of the UV_THREADPOOL_SIZE, we need to use more THREADs.

### What is the feature you are proposing to solve the problem?

Add support of the UV_THREADPOOL_SIZE, we need to use more THREADs.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fixes #14398

more threads

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit ee6b740514d9c6981c78068166eb06a783a4c4a3 is building: [`#4516`](https://buildkite.com/bun/bun/builds/4516)

File: src/bun.zig [hunk 1]
```diff
@@ -3928,3 +3928,34 @@ pub fn indexOfPointerInSlice(comptime T: type, slice: []const T, item: *const T)
...
+    const max_threads = 1024;
+    const min_threads = 2;
+    const ThreadCount = struct {
+        pub var cached_thread_count: u16 = 0;
+        var cached_thread_count_once = std.once(getThreadCountOnce);
+        fn getThreadCountFromUser() ?u16 {
+            inline for (.{ "UV_THREADPOOL_SIZE", "GOMAXPROCS" }) |envname| {
+                if (getenvZ(envname)) |env| {
+                    if (std.fmt.parseInt(u16, env, 10) catch null) |parsed| {
+                        if (parsed >= min_threads) {
```
Comment (dylan-conway) [inline src/bun.zig:3952]: GOMAXPROCS=1 is a thing right? Should min_threads be 1 instead of 2?

Comment (Jarred-Sumner) [reply src/bun.zig:3952]: it should be 2

Comment (Jarred-Sumner) [reply src/bun.zig:3952]: sometimes we depend on threads to exist

**Pull request merged**